src/client/entity-editor/author-credit-editor/author-credit-row.tsx

```diff
@@ -80,22 +84,42 @@ function AuthorCreditRow({
 	author,
 	joinPhrase,
 	name,
+	isUf,
 	onAuthorChange,
 	onJoinPhraseChange,
+	onClearHandler,
 	onNameChange,
-	onRemoveButtonClick
+	onRemoveButtonClick,
+	...rest
 }: Props) {
+	const SelectWrapper = !isUf ? EntitySearchFieldOption : SearchEntityCreate;
+	const onChangeHandler = React.useCallback((value, action) => {
+		if (action && ['clear', 'pop-value', 'select-option'].includes(action.action) && author && author.get('__isNew__', false)) {
+			onClearHandler(author.get('id'));
+		}
+		onAuthorChange(value);
+	}, [author]);
+	const handleButtonClick = React.useCallback(() => {
+		// don't remove author if it's first row
+		if (index !== 'n0' && author?.get('__isNew__')) {
+			onClearHandler(author.get('id'));
+		}
+		onRemoveButtonClick();
+	}, [author, index]);
 	return (
 		<div>
 			<Row>
 				<Col md={{span: 3}}>
-					<EntitySearchFieldOption
+					<SelectWrapper
 						instanceId={`author${index}`}
+						isUf={isUf}
 						label="Author"
+						rowId={index}
 						type="author"
 						validationState={!author ? 'error' : null}
 						value={author}
-						onChange={onAuthorChange}
+						onChange={onChangeHandler}
+						{...rest}
```

Comment:
Some props like entity identifiers and types are required for modal, since these props required all over the form components so thought this would be better rather than manually cherry picking the props. These props don't get change so should not cause issue with renders.